Sources/PrivoSDK/PrivoAuth.swift

```diff
@@ -237,16 +237,32 @@ public class PrivoAuth {
                 try .init(child: child)
             ])
         let response = try await api.registerParentAndChild(parentChildPair, gwToken)
+        var resetPasswordLink = response.to.updatePasswordLink
         
-        return .init(resetPasswordLink: response.to.updatePasswordLink)
+        let serviceIdentifier = PrivoInternal.settings.serviceIdentifier
+        if let serviceInfo = await api.getServiceInfo(serviceIdentifier: serviceIdentifier),
+           let siteId = serviceInfo.p2siteId
+        {
+            var components = resetPasswordLink.urlComponent()
+            let queryItemsOld = components.queryItems ?? []
+            let queryItemsNoSiteId = queryItemsOld.filter {
```

Comment:
how about moving this logic to the extension?

Reply:
done